roles/manage-repositories/tasks/enable_repositories.yml

```diff
@@ -0,0 +1,31 @@
+---
+# tasks file for manage-repositories
+- name: "Enable all the required repositories for Satellite6 Installation."
+  when: (satellite_distribution == "internal_repo") or
+        (satellite_distribution == "cdn")
+  rhsm_repository:
+    name: "{{ item }}"
+    state: enabled
+  with_items:
+     - "rhel-{{ ansible_distribution_major_version }}-server-rpms"
+     - "rhel-{{ ansible_distribution_major_version }}-server-optional-rpms"
+     - "rhel-{{ ansible_distribution_major_version }}-server-extras-rpms"
+     - "rhel-{{ ansible_distribution_major_version }}-server-satellite-maintenance-6-rpms"
+     - "rhel-{{ ansible_distribution_major_version }}-server-ansible-2-rpms"
+     - "rhel-server-rhscl-{{ ansible_distribution_major_version }}-rpms"
+
+- name: "Enable Satellite6 repository."
+  when:
+    - satellite_distribution == "cdn"
+  rhsm_repository:
+    name: "rhel-{{ ansible_distribution_major_version }}-server-satellite-{{ satellite_version }}-rpms"
+    state: enabled
+
+- name: "Add Satellite 6 repository"
+  when:
+    - satellite_distribution == "internal_repo"
+  yum_repository:
+    name: "{{ satellite_reponame }}"
```

Comment:
Best practices say to name your variables as scoped by the role itself to prevent clashes. So this would translate to `manage_repositories_satellite_reponame`